odoo/tests/common.py

```diff
@@ -462,6 +462,52 @@ def get_unaccent_wrapper(cr):
                 "\n---- actual query:\n%s\n---- not like:\n%s" % (actual_query, expect_query),
             )
 
+    @contextmanager
+    def assertQuery(self, expected, flush=True):
+        Cursor_execute = Cursor.execute
+        actual_queries = []
+
+        def execute(self, query, params=None, log_exceptions=None):
+            actual_queries.append(query)
+            return Cursor_execute(self, query, params, log_exceptions)
+
+        def get_unaccent_wrapper(cr):
+            return lambda x: x
+
+        def as_chars(query: str):
+            result = []
+            query = query.lower()
+            for c in query:
+                if not c.isspace():
+                    result.append(c)
+            return result
```

Comment:
> Errr, isn't this simply query.lower().split()?

Almost, the subtility comes from delimitors, in this particular case the returned query by the ORM is something like:

```sql
select foo, bar , baz from my_table
```
if you notice the inconsistency with the commas, the call to `.lower().split()` will return `['select', 'foo,', 'bar', ',', ...]`

if I write my query like
```sql
select foo, bar, baz from my_table
```
they will never match despite an inconsequential difference, in fact this is exactly what happened when I wrote my query by hand then compared it to what the ORM returned.

By converting the query into a list of non-whitespace chars, the problem is solved.

I realize I'm being kinda anal about it, I could've just changed the formatting of my delimiters but I figured "eh why not".
> Changing a = b to a=b will make your method crash... for nothing critical

I don't understand, the result would be ['a', '=', 'b'] for both, no?

Reply:
Okay, what the function returns is in fact `list("".join(query.lower().split()))`.
So what you do is 100% equivalent to the string comparison done in `assertQueries()`.

Reply:
You're right, I probably fucked up while copying the code from `assertQueries` into this one and ommitted the `"".join()` which indeed serves the same purpose.

With that out of the way, I refactored this so that both methods can coexist with minimal code duplication, let me know what you think, if we should keep it as-is or if we should simply merge them into a single method.